PyRDF/backend/Dist.py

```diff
@@ -240,41 +228,104 @@ def _get_clustered_ranges(self, nentries, treename, filelist,
             warnings.warn(msg, UserWarning, stacklevel=2)
             self.npartitions = numclusters
 
-        partSize = numclusters // self.npartitions
-        remainder = numclusters % self.npartitions
+        logger.debug("{} clusters will be split along {} partitions.".format(
+            numclusters, self.npartitions))
 
-        i = 0  # Iterator
-        ranges = []
-        entries_to_process = 0
-
-        while i < numclusters:
-            index_start = i
-            start = clusters[i][0]
-            i = i + partSize
-            if remainder > 0:
-                i += 1
-                remainder -= 1
-            index_end = i
-            if i == numclusters:
-                end = clusters[-1][1]
-            else:
-                end = clusters[i - 1][1]
+        """
+        Split list of clusters into multiple sublists. Each sublist will hold
+        the clusters that should fit into each partition of the distributed
+        dataset, e.g.
+        [
+            # Partition 1 will process the following clusters
+            [
+                (start_0, end_0, offset_0, (filename_0, fileindex_0)),
+                (start_1, end_1, offset_1, (filename_1, fileindex_1)),
+                ...,
+                (start_n, end_n, offset_n, (filename_n, fileindex_n))
+            ],
+            # Partition 2 will process these other clusters
+            [
+                (start_n+1, end_n+1, offset_n+1, (filename_n+1, fileindex_n+1)),
+                ...,
+                (start_m, end_m, offset_m, (filename_m, fileindex_m))
+            ],
+            ...
+        ]
+        """
+        clusters_split_by_partition = list(
+            _n_even_chunks(filesandclusters, self.npartitions))
+
+        logger.debug("Number of clusters per partition: {}".format([
+            len(clusters) for clusters in clusters_split_by_partition
+        ]))
+
+        partitions_startentries = [min(clusters)[0]
```

Comment:
Why `max(clusters)[0]`, I thought `max` always returns a scalar value?

Reply:
If you have tuples in the list `max` and `min` are computed along tuple indexes:
```
>>> a = [(0,99),(100,199),(200,299)]
>>> max(a)
(200, 299)
>>> min(a)
(0, 99)
>>> 
```

Reply:
Interesting, didn't know that it just returns the tuple with largest element!

Reply:
For posterity, Python sorts by the first item of each tuple, and breaks any ties on the second. This generalises to items with > 2 dimensions :)

Reply:
Hi @agoose77,
Thank you for the input :D Indeed in this case we are dealing with sequential tuples so there's no risk of ties. Good to know it nonetheless!